### PR TITLE
feat(static_storage): Add support for whitenoise (optional)

### LINKED ISSUE #213: Add support for whitenoise
Sometimes we just don't need the weight of AWS for media assets. I think this should be an option in the cookiecutter.json file.

### PR DESCRIPTION
**Pull request by theskumar**
- Add option to opt for whitenoise via cookiecutter
- Seperate Media and Static Configuration
- https://whitenoise.readthedocs.org/

closes #213

### CODE REVIEW

**Comment by theskumar** [comment]
:bell: @pydanny @luzfcb review me please! :)

File: {{cookiecutter.repo_name}}/config/settings/production.py [hunk 1]
```diff
@@ -85,8 +81,22 @@
...
 }
 
-# See: https://docs.djangoproject.com/en/dev/ref/settings/#static-url
-STATIC_URL = 'https://s3.amazonaws.com/%s/' % AWS_STORAGE_BUCKET_NAME
+# URL that handles the media served from MEDIA_ROOT, used for managing stored files.
+MEDIA_URL = 'https://s3.amazonaws.com/%s/' % AWS_STORAGE_BUCKET_NAME
+
+# Static Assests
+# ------------------------
+{% if cookiecutter.use_whitenoise == 'y' -%}
```
Comment (luzfcb) [inline {{cookiecutter.repo_name}}/config/settings/production.py]: There are some typos, see ``` - ``` caracter in ```-%}```

File: {{cookiecutter.repo_name}}/config/settings/production.py [hunk 2]
```diff
@@ -85,8 +81,22 @@
...
+STATICFILES_STORAGE = 'whitenoise.django.GzipManifestStaticFilesStorage'
+{% else %}
+STATICFILES_STORAGE = DEFAULT_FILE_STORAGE
+STATIC_URL = MEDIA_URL
+
+# See: https://github.com/antonagestam/collectfast
+# For Django 1.7+, 'collectfast' should come before 'django.contrib.staticfiles'
+AWS_PRELOAD_METADATA = True
+INSTALLED_APPS = ('collectfast', ) + INSTALLED_APPS
+{%- endif %}
```
Comment (luzfcb) [inline {{cookiecutter.repo_name}}/config/settings/production.py]: Same as above, see ``` - ``` caracter in ```{%-```

File: {{cookiecutter.repo_name}}/config/wsgi.py [hunk 3]
```diff
@@ -15,6 +15,11 @@
 """
 import os
 
+from django.core.wsgi import get_wsgi_application
+{% if cookiecutter.use_whitenoise == 'y' -%}
```
Comment (luzfcb) [inline {{cookiecutter.repo_name}}/config/wsgi.py]: There are some typos, see - caracter in -%}

File: {{cookiecutter.repo_name}}/config/wsgi.py [hunk 4]
```diff
@@ -24,9 +29,14 @@
...
 # file. This includes Django's development server, if the WSGI_APPLICATION
 # setting points here.
-from django.core.wsgi import get_wsgi_application  # noqa
 application = get_wsgi_application()
 
+{% if cookiecutter.use_whitenoise == 'y' -%}
+# Use Whitenoise to serve static files
+# See: https://whitenoise.readthedocs.org/
+application = DjangoWhiteNoise(application)
+{%- endif %}
```
Comment (luzfcb) [inline {{cookiecutter.repo_name}}/config/wsgi.py]: Same as above, see ```-``` caracter in ```{%-```

File: {{cookiecutter.repo_name}}/requirements/production.txt [hunk 5]
```diff
@@ -2,8 +2,17 @@
...
+# WSGI Handler
+# ------------------------------------------------
+gevent==1.0.1
 gunicorn==19.3.0
+
+# Static and Media Storage
+# ------------------------------------------------
+boto==2.38.0
 django-storages-redux==1.2.3
+{% if cookiecutter.use_whitenoise == 'y' -%}
```
Comment (luzfcb) [inline {{cookiecutter.repo_name}}/requirements/production.txt]: Same as above

File: {{cookiecutter.repo_name}}/requirements/production.txt [hunk 6]
```diff
@@ -2,8 +2,17 @@
...
+whitenoise==1.0.6
+{% else %}
 Collectfast==0.2.3
-gevent==1.0.1
-boto==2.38.0
+{%- endif %}
```
Comment (luzfcb) [inline {{cookiecutter.repo_name}}/requirements/production.txt]: Same as above

**Comment by theskumar** [comment]
@luzfcb these are not typos, they are jinja2 feature for whitespace control[1]. 

[1] http://jinja.pocoo.org/docs/dev/templates/#whitespace-control

**Comment by luzfcb** [comment]
@theskumar oh, I swear I tried this in the documentation. :-1: to me. :-P . a big, huge apology. 
thank you for pointing out my mistake, and teach me something new.

**Comment by theskumar** [comment]
@luzfcb haha! no worries! This is something I learnt myself very recently while having discussion about new jinja2 support in django with one of my fellow developer on twitter[1]. Jinja2 gets many things right! 

[1] https://twitter.com/kracetheking/status/583504889289940992

**Pull request merged**